common/src/main/java/org/apache/celeborn/common/network/server/MapDataPartition.java

```diff
@@ -0,0 +1,274 @@
+  public static final Logger logger = LoggerFactory.getLogger(MapDataPartition.class);
+  private final FileInfo fileInfo;
+  private final ExecutorService readExecutor;
+  private final ConcurrentHashMap<Long, MapDataPartitionReader> readers =
+      JavaUtils.newConcurrentHashMap();
+  private FileChannel dataFileChanel;
+  private FileChannel indexChannel;
+  private long indexSize;
+  private volatile boolean isReleased = false;
+  private final BufferQueue bufferQueue = new BufferQueue();
+  private boolean bufferQueueInitialized = false;
+  private MemoryManager memoryManager = MemoryManager.instance();
+  private Consumer<Long> recycleStream;
+  private int minReadBuffers;
+  private int maxReadBuffers;
+  private int fileBuffers;
+  private int minBuffersToTriggerRead;
+
+  public MapDataPartition(
+      int minReadBuffers,
+      int maxReadBuffers,
+      HashMap<String, ExecutorService> storageFetcherPool,
+      int threadsPerMountPoint,
+      FileInfo fileInfo,
+      Consumer<Long> recycleStream,
+      int minBuffersToTriggerRead)
+      throws IOException {
+    this.recycleStream = recycleStream;
+    this.fileInfo = fileInfo;
+
+    this.minReadBuffers = minReadBuffers;
+    this.maxReadBuffers = maxReadBuffers;
+    this.fileBuffers = (int) Math.ceil(fileInfo.getFileSize() * 1.0 / fileInfo.getBufferSize());
+
+    updateBuffersTarget((this.minReadBuffers + this.maxReadBuffers) / 2 + 1);
+
+    logger.debug(
+        "read map partition {} with {} {} {}",
+        fileInfo.getFilePath(),
+        bufferQueue.getLocalBuffersTarget(),
+        fileInfo.getBufferSize());
+
+    this.minBuffersToTriggerRead = minBuffersToTriggerRead;
+
+    readExecutor =
+        storageFetcherPool.computeIfAbsent(
+            fileInfo.getMountPoint(),
+            k ->
+                Executors.newFixedThreadPool(
+                    threadsPerMountPoint,
+                    new ThreadFactoryBuilder()
+                        .setNameFormat(fileInfo.getMountPoint() + "-reader-thread-%d")
+                        .setUncaughtExceptionHandler(
+                            (t1, t2) -> {
+                              logger.warn("StorageFetcherPool thread:{}:{}", t1, t2);
+                            })
+                        .build()));
+    this.dataFileChanel = new FileInputStream(fileInfo.getFile()).getChannel();
+    this.indexChannel = new FileInputStream(fileInfo.getIndexPath()).getChannel();
+    this.indexSize = indexChannel.size();
+
+    MemoryManager.instance().addReadBufferTargetChangeListener(this);
+  }
+
+  private synchronized void updateBuffersTarget(int buffersTarget) {
+    int currentBuffersTarget = buffersTarget;
+    if (currentBuffersTarget < minReadBuffers) {
+      currentBuffersTarget = minReadBuffers;
+    }
+    if (currentBuffersTarget > maxReadBuffers) {
+      currentBuffersTarget = maxReadBuffers;
+    }
+    if (currentBuffersTarget > fileBuffers) {
```

Comment:
when data is broadcast to downstream, the file size may smaller, but actually we need more file buffers, probably to readersize * fileBuffers, so we need know sub partition number. And besides we actually need more buffers for endOfDataevent/PartitionEvent(these need buffers too). May be we just remove fileBuffers this time?